### ISSUE TITLE
Error mounting azure file share

### ISSUE THREAD

**Issue by joshlester** [Azure/azurefile-dockervolumedriver#47]
Hi,

I'm running docker version 1.12.0-beta21 (build: 10868) on osx: 10.11.6.

After following the instruction here:
https://github.com/Azure/azurefile-dockervolumedriver/tree/master/contrib/init/systemd

And creating a share successfully using the following command:
_docker volume create -d azurefile -o share=rendivo --name rendivo_

When running the command: 
_docker run --name comp -p 80:80 -d -v rendivo:/media/rendivo 3dviz/comp-rendivo_

I get the following error:

```
_docker: Error response from daemon: VolumeDriver.Mount: mount failed: exit status 1
output="mount error: could not resolve address for vhdslvtw04vcgddhrgkha4kt.file.: Unknown error\n"._

After running the command _journalctl -fu azurefile-dockervolumedriver_ I get the following error messages:

_-- Logs begin at Tue 2016-08-02 03:31:31 UTC. --
Aug 02 03:51:07 host-comp-rendivo systemd[1]: Started Azure File Service Docker Volume Driver.
Aug 02 04:09:36 host-comp-rendivo azurefile-dockervolumedriver[5068]: time="2016-08-02T04:09:36Z" level=error msg="could not fetch metadata: cannot read metadata: open /etc/docker/plugins/azurefile/volumes/rendivo: no such file or directory" name=rendivo operation=get
Aug 02 04:09:36 host-comp-rendivo azurefile-dockervolumedriver[5068]: time="2016-08-02T04:09:36Z" level=info msg="created azure file share \"rendivo\"" name=rendivo operation=create options=map[share:rendivo]
Aug 02 04:21:03 host-comp-rendivo azurefile-dockervolumedriver[5068]: time="2016-08-02T04:21:03Z" level=error msg="mount failed: exit status 1\noutput=\"mount error: could not resolve address for vhdslvtw04vcgddhrgkha4kt.file.: Unknown error\\n\"" name=rendivo operation=mount
Aug 02 04:44:25 host-comp-rendivo azurefile-dockervolumedriver[5068]: time="2016-08-02T04:44:25Z" level=error msg="mount failed: exit status 1\noutput=\"mount error: could not resolve address for vhdslvtw04vcgddhrgkha4kt.file.: Unknown error\\n\"" name=rendivo operation=mount
Aug 02 04:49:14 host-comp-rendivo azurefile-dockervolumedriver[5068]: time="2016-08-02T04:49:14Z" level=error msg="could not fetch metadata: cannot read metadata: open /etc/docker/plugins/azurefile/volumes/rendivo: no such file or directory" name=rendivo operation=get
Aug 02 04:49:14 host-comp-rendivo azurefile-dockervolumedriver[5068]: time="2016-08-02T04:49:14Z" level=info msg="created azure file share \"rendivo\"" name=rendivo operation=create options=map[share:rendivo]
Aug 02 04:50:54 host-comp-rendivo azurefile-dockervolumedriver[5068]: time="2016-08-02T04:50:54Z" level=error msg="mount failed: exit status 1\noutput=\"mount error: could not resolve address for vhdslvtw04vcgddhrgkha4kt.file.: Unknown error\\n\"" name=rendivo operation=mount_
```

Any help is appreciated.
Thanks,
Josh.

**Comment by ahmetb** [comment]
@joshlester it appears like we might have introduced a bug in the last release. Do you mind sending your `azurefile-dockervolumedriver.default` file? I will try to reproduce it.

By the way, I see you specified OS X above, are you running "Docker for Mac"?

**Comment by stefanegg** [comment]
@ahmetalpbalkan While I was trying to follow up on another issue (#45) using the latest Docker and azurefile-dockervolumedriver releases, I experienced the same issues as Josh describes above. Using Docker 1.12.0, azurefile-dockervolumedriver 0.4.0 and a Ubuntu 16.04 LTS VM on Azure. I also tried with and without the newly introduced AZURE_STORAGE_BASE option in the .default file.

**Comment by ahmetb** [comment]
argh, that's weird. If the option is empty string it should be using the default value correctly. I will give it a try. Thanks for reporting.

**Comment by ahmetb** [comment]
I can reproduce the bug. Working on it.

**Comment by ahmetb** [comment]
Hi folks just released version 0.4.1 with the fix. Thanks for reporting this again!

**Comment by joshlester** [comment]
Wow, thanks for the quick response! I can confirm it's now working on my machine.
